src/main/java/com/google/devtools/build/lib/remote/ByteStreamBuildEventArtifactUploader.java

```diff
@@ -236,7 +237,9 @@ private Single<List<PathMetadata>> queryRemoteCache(
     List<PathMetadata> filesToQuery = new ArrayList<>();
     Set<Digest> digestsToQuery = new HashSet<>();
     for (PathMetadata path : paths) {
-      if (shouldQuery(path)) {
+      if (!shouldUploadBasedOnUploadMode(path)) {
```

Comment:
Maybe move this check into `shouldQuery`. Tests are failing because we didn't add path into `knownRemotePaths` below.

Reply:
The reason I didn't put it into `shouldQuery`, was because the path might not be remote. So not sure on the answer to this.